eng/pipelines/libraries/build-job.yml

```diff
@@ -81,27 +81,102 @@ jobs:
         - ${{ if eq(parameters.useHelix, true) }}:
           - _additionalBuildArguments: /p:ArchiveTests=true
 
+        # Currently only support pipeline caching for PRs and manual triggered runs
+        - ${{ if or(in(variables['Build.Reason'], 'PullRequest'), in(variables['Build.Reason'], 'Manual')) }}:
+          - name: UsePipelineCache
+            value: true
+
+        - name: ConfigurationName
+          value: ${{ format('Libraries {0} {1}{2} {3} {4} {5}', parameters.runtimeVariant, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig, parameters.framework) }}
+
+        - name: CommitsOutputFile
+          value: '$(Build.SourcesDirectory)/libraries.commit_hash'
+
+        - name: GetCommitScript
+          value: '$(Build.SourcesDirectory)/src/coreclr/scripts/get_commit_hash.py'
+
+        # Otherwise set all the cache-specific variables to 'false' which will ensure to build product.
+        - ${{ if and(notin(variables['Build.Reason'], 'PullRequest'), notin(variables['Build.Reason'], 'Manual')) }}:
```

Comment:
Can we put all these variables that are shared across different templates on xplat-setup which is used by all templates?